design/baremetal-operator/bmc-events.md

```diff
@@ -0,0 +1,101 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Event Subscription API
+
+Users of bare metal hardware may want to receive events from the
+baseboard management controller (BMC) in order to act on them, such as
+failed hardware, increases in temperature, and other reasons.
+
+## Status
+
+provisional
+
+## Summary
+
+The Redfish standard includes the ability to subscribe to events, which
+will cause hardware events to be sent in a particular format to a target
+URI. This design document describes a Metal3 API for configuring a
+subscription. While Redfish is the primary target for this design, the
+Ironic API is vendor-neutral and seeks to provide a unified interface
+for configuring events.
+
+## Motivation
+
+Some environments run workloads that need to deal with potential
+failures or environmental changes quicker than they would get an alert
+from other kind of eventing software. For example, some workloads may
+have a sidecar container that knows how to deal with an alert that a
+particular network interface went down, or that the CPU temperature
+reached a certain threshold.
+
+### Goals
+
+- Provide an API to manage subscriptions to events
+
+### Non-Goals
+
+- Any kind of event polling
+- Software for processing the events, i.e. any webhook
+- BMC's beyond Redfish for now
+
+## Proposal
+
+### User Stories
+
+- As a user, I'd like to configure a my BMC to send events to a target URL.
+- As a user, I'd like to filter the types of events the BMC sends to my target URL.
+- As a user, I'd like to provide context to a particular event subscription.
+- As a user, I'd like to provide arbitrary headers.
+
+## Design Details
+
+### Implementation Details
+
+![Sketch of BMCEventSubscription API](./bmc-events.png)
```

Comment:
This will be easier to iterate on if it's inlined in the text instead of using an image.

Reply:
Done